File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        freqMap={}
        n = len(nums)
        keys=[]

        for num in nums:
            if num in freqMap:
                freqMap[num]+=1
            else:
                freqMap[num] = 1
        
        for key in freqMap:
            keys.append(key)

        for j in range(len(keys)):
            nums[j] = keys[j]

        return len(keys)        
        

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.c
int removeDuplicates(int* nums, int numsSize) {
    int i = 0;

    while (i < numsSize - 1) {
        if (nums[i] == nums[i + 1]) {
            // delete nums[i+1] by shifting left
            for (int j = i + 1; j < numsSize - 1; j++) {
                nums[j] = nums[j + 1];
            }
            numsSize--;
        } else {
            i++;
        }
    }
    return numsSize;
}
